cocos/base/ZipUtils.cpp

```diff
@@ -609,6 +636,37 @@ bool ZipFile::fileExists(const std::string &fileName) const
     return ret;
 }
 
+const std::vector<std::string> ZipFile::listFiles(const std::string &pathname) const
+{
+
+    // filter files which `filename.startsWith(pathname)`
+    // then make each path unique
+
+    std::set<std::string> fileSet;
+    ZipFilePrivate::FileListContainer::const_iterator it = _data->fileList.begin();
+    ZipFilePrivate::FileListContainer::const_iterator end = _data->fileList.end();
+    //ensure pathname ends with `/` as a directory
+    std::string dirname = pathname[pathname.length() -1] == '/' ? pathname : pathname + "/";
+    while(it != end)
+    {
+        const std::string &filename = it->first;
+        if(filename.substr(0, dirname.length()) == dirname)
+        {
+            std::string suffix = filename.substr(dirname.length());
+            const std::vector<std::string>& parts = splitFilePath(suffix);
+            if(parts.size() == 1)
+            {
+                fileSet.insert(suffix);
+            }else if(parts.size() > 1){
+                fileSet.insert(parts[0] + "/");
+            }
```

Comment:
`splitPath` is too complex here, we don't need to know every part content, just need to check if it has `/` or not?

Reply:
yes.